eq/qt/windowFactory.cpp

```diff
@@ -20,15 +20,46 @@
 #include "window.h"
 #include "windowImpl.h"
 
+#include "eq/pipe.h"
+
+#include <QGuiApplication>
+
 namespace eq
 {
 namespace qt
 {
 
-detail::Window* WindowFactory::onCreateImpl( const WindowSettings& settings,
+detail::Window* WindowFactory::onCreateImpl( const eq::Pipe* pipe,
+                                             const WindowSettings& settings,
                                              QThread* thread_ )
 {
-    return Window::createImpl( settings, thread_ );
+    // Trying to infer the screen to use from the pipe device number.
+    // We will simply assume that each QScreen belongs a display and that
+    // each GPU drives a different display. This will work for most setups,
+    // but will break at the moment a single GPU is driving more than one
+    // screen. The only solution in those cases is to adjust manually the
```

Comment:
You can enforce it with a custom X server configuration file. It's not the default for sure, but it's not impossible.